### ISSUE TITLE
Cleanup Code

### ISSUE THREAD

**Issue by hgn** [protocollabs/dmpr-simulator#15]
The code should be cleaned up in some way. It is hard to follow what can be executed, what can be analysed, how can the scenarios be executed and what particular analyser can be used on what data - without reading to the source and and all files.

So I have no clue to do this in a clean way. First of all, the makefile should be a "convenience wrapper" for a python script. The python script should be callable by itself and should print out all possibilities. The analyzer script should be called automatically by this script or by hand. But it should be obvious that these script can, after a certain test run, called. If this is not possible then the python script should call these scripts. Every other python script, not callable by the user, like draw.py should be put into a subdirectory.

Is __init__.py really needed in our case?

 size_combinations.checkpoint should be put/generated into a tmp directory so that it is obvious that these file is somehow tmp.

The makefile should have a target where every generated file is remove (make clean). 

The analyze directory is somehow complex. Is it to store the analyzer scripts? Is it for storing results?
I prefer to save all python relevant code somehow in a src directory. So that a user know, "no, this is a source dir, I don't want to fix/extend the source, I just want to run the simulation".

`topologies` and `scenarios` are fine. They "communicate": "yes If you want to extend the topologies or scenarios, use me". But there are files in it where I excepted to see them in the src folder too. E.g. circle_seqdiag.py and friends.

So at the end I think the directory structure, file placement and user experience should be changed.

**Comment by reisub-de** [comment]
- [ ] Improve Makefile, all, clean
- [ ] split modules and executables
- [ ] directory for results and other stuff
- [ ] __init__.py ?

**Comment by reisub-de** [comment]
- [ ] pypy and deps: pypy is faster but is problematic with matplotlib/numpy/cairo/pil -> Solution needed for simulation and analyze in same call (subprocess? ugly!)

**Comment by hgn** [comment]
mhh, I like the current solution in the makefile. I think we *should* focus on cpython and using pypy is somehow a special setup and manual intervention is required. The cpython way should work cleanly without any intervention at all. mmhh not sure how to address.

the subprocess way is probably not so bad. We can provide a wrapper somewhere, lets call them pyexec() and this method hides pypy checking and fallback functionality. as I said, not sure

**Comment by reisub-de** [comment]
I already found a fix. matplotlib, numpy and pillow (pil) are compatible with pypy and cairo can be directly exchanged for cairocffi which is compatible with python *and* pypy.

Therefore we don't make any python interpreter selection in the makefile and let it be the users choice to enable a venv with pypy beforehand if they want to use it.

**Comment by hgn** [comment]
perfect!
